superdesk/core/auth/privilege_rules.py

```diff
@@ -0,0 +1,62 @@
+from typing import Any, cast
+
+from quart_babel import gettext
+from superdesk.errors import SuperdeskApiError
+from superdesk.users.async_service import get_privileges, is_admin
+from superdesk.core.types import HTTP_METHOD, AuthRule, Request
+
+
+def required_privilege_rule(privilege: str) -> AuthRule:
+    """
+    Creates an authentication rule that restricts access based on a specific privilege.
+
+    Args:
+        privilege (str): The privilege required to access the resource.
+
+    Returns:
+        AuthRule: An asynchronous rule function that checks if the user has the required privilege.
+
+    The rule verifies:
+    - If the user is an admin, access is granted without further checks.
+    - Otherwise, the user's privileges are retrieved and checked against the required privilege.
+
+    Raises:
+        SuperdeskApiError: If the user does not have the required privilege.
+    """
+
+    async def internal_rule(request: Request):
+        current_user = request.user
+
+        # TODO-ASYNC: Check if for admin we need to pull all privileges instead
+        if is_admin(current_user):
+            return None
+
+        user_role = request.storage.request.get("role")
+        user_privileges = get_privileges(request.user, user_role)
+
+        if user_privileges.get(privilege, False):
+            return None
+
+        raise SuperdeskApiError.forbiddenError(message=gettext("Insufficient privileges for the requested operation."))
+
+    return internal_rule
+
+
+def privilege_based_rules(rules: dict[HTTP_METHOD, str]) -> dict[str, AuthRule]:
```

Comment:
As this rule is for checking against a HTTP method / role dictionary, it might be good to indicate that in the name of the rule